test/unit/Git/FetchAndSetCurrentUserByReplacingCurrentOriginRemoteTest.php

```diff
@@ -105,4 +106,27 @@ public function testFetchesAndSetsCurrentUserWithoutOrigin(): void
         self::assertStringContainsString('origin/initial-branch', $fetchedBranches);
         self::assertStringContainsString('origin/new-branch', $fetchedBranches);
     }
+
+    public function testFailingToSetARemoteWillNotLeadToSecretExfiltration(): void
+    {
+        $sourceUri            = $this->createMock(UriInterface::class);
+        $sourceUriWithSecrets = $this->createMock(UriInterface::class);
+
+        $sourceUri->method('__toString')
+            ->willReturn('https://github.com/laminas/this-repository-does-not-exist.git');
+        $sourceUriWithSecrets->method('__toString')
+            ->willReturn('https://SUPERSECRET:x-oauth-basic@github.com/laminas/this-repository-does-not-exist.git');
+
+        $notAGitDirectory = Filesystem\create_temporary_file();
+
+        Filesystem\delete_file($notAGitDirectory);
+        Filesystem\create_directory($notAGitDirectory);
+
+        try {
+            (new FetchAndSetCurrentUserByReplacingCurrentOriginRemote($this->variables))
+                ->__invoke($sourceUri, $sourceUriWithSecrets, $notAGitDirectory);
+        } catch (Throwable $failure) {
+            self::assertDoesNotMatchRegularExpression('/SUPERSECRET/m', $failure->getMessage());
+        }
+    }
```

Comment:
Verified locally that this test fails on `1.14.x`, while succeeds on this branch.